src/core/server/kibana_config.ts

```diff
@@ -18,9 +18,22 @@
  */
 
 import { schema, TypeOf } from '@kbn/config-schema';
+import { ConfigDeprecationProvider } from '@kbn/config';
 
 export type KibanaConfigType = TypeOf<typeof config.schema>;
 
+const deprecations: ConfigDeprecationProvider = () => [
+  (settings, fromPath, log) => {
+    const kibana = settings[fromPath];
+    if (kibana?.index) {
+      log(
+        `Setting [kibana.index] is deprecated. Multitenancy by changing 'kibana.index' will not be supported starting in 8.0. See https://ela.st/kbn-remove-legacy-multitenancy for more details`
+      );
```

Comment:
Remark: There is an inconsistency with the wording of our `unused` factory:

https://github.com/elastic/kibana/blob/5710f6763b69762bd79c0f3ecdc46fc7e3179627/packages/kbn-config/src/deprecation/deprecation_factory.ts#L90

`Setting [{config prop}] is deprecated` vs `{config prop} is deprecated`. 

I also see that `rename` is using double quote around the config prop (where `unused` does not). 

https://github.com/elastic/kibana/blob/5710f6763b69762bd79c0f3ecdc46fc7e3179627/packages/kbn-config/src/deprecation/deprecation_factory.ts#L47

This is minor, but we might want to be consistent in our deprecation messages.

Also thinking: we may want to adapt `ConfigDeprecationFactory.unused` (and others) to accept a new optional 'message' parameter. That would also to log custom messages in deprecations without having to recode a custom one just to log the message (as it is done here for `kibana.index`). 

`unused('kibana.index', "Multitenancy by changing 'kibana.index' will not be supported starting in 8.0. See https://ela.st/kbn-remove-legacy-multitenancy for more details")`

This is probably out of the scope of this PR though. Should I create a separate issue?

Reply:
> This is minor, but we might want to be consistent in our deprecation messages.

Agreed about being consistent here. I was originally replicating the format from the following deprecation log message, because I incorrectly assumed it used the "standard approach": `[warning][config][deprecation] Setting [elasticsearch.username] to "kibana" is deprecated. You should use the "kibana_system" user instead.`

Do we want to follow the precedent set by `unused` and not use double-quotes, or should we follow the precedent set by `rename` and use double-quotes?

> Also thinking: we may want to adapt ConfigDeprecationFactory.unused (and others) to accept a new optional 'message' parameter. That would also to log custom messages in deprecations without having to recode a custom one just to log the message (as it is done here for kibana.index).

I think it's fine to support custom messages for `unused`/`rename`.  However, I don't think we should be using the `unused` deprecation here, as this setting is still being used and has an effect, it's just deprecated. We use the `unused` deprecation elsewhere when settings don't have an effect.

Reply:
> Do we want to follow the precedent set by unused and not use double-quotes, or should we follow the precedent set by rename and use double-quotes?

I think using quotes is slightly better, wdyt?

> I think it's fine to support custom messages for unused/rename. However, I don't think we should be using the unused deprecation here, as this setting is still being used and has an effect, it's just deprecated.

Yea, you're right. I forgot we are still using the value of this property. Nevermind

Reply:
Double-quotes sound good to me! I just independently decided that also and used it in https://github.com/elastic/kibana/pull/83988/commits/9dc7532e4c1441994a2323e37fec020f38915541